tests/integration/test_active_response/test_analysisd/test_os_exec.py

```diff
@@ -0,0 +1,351 @@
+            'TIMEOUT': 0
+        },
+        'metadata': {
+            'agents_number': 2,
+            'agents_os': ['debian8', 'ubuntu20.04'],
+            'agents_version': ['v4.1.0', 'v4.2.0'],
+            'protocol': 'tcp',
+            'extra_args': 'no',
+            'timeout': 'no',
+            'all_agents': 'yes'
+        }
+    }
+]
+
+params = [case['params'] for case in cases]
+metadata = [case['metadata'] for case in cases]
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+configurations = load_wazuh_configurations(configurations_path, __name__, params=params, metadata=metadata)
+
+# List where the agents objects will be stored
+agents = []
+
+
+@pytest.fixture(scope="session")
+def set_debug_mode():
+    local_int_conf_path = os.path.join(WAZUH_PATH, 'etc', 'local_internal_options.conf')
+    debug_line = 'remoted.debug=2\n'
+    with open(local_int_conf_path, 'r') as local_file_read:
+        lines = local_file_read.readlines()
+        for line in lines:
+            if line == debug_line:
+                return
+    with open(local_int_conf_path, 'a') as local_file_write:
+        local_file_write.write('\n'+debug_line)
+
+
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    yield request.param
+
+
+@pytest.fixture(scope="function")
+def restart_service():
+    clean_logs()
+    control_service('restart')
+    yield
+
+
+@pytest.fixture(scope="function")
+def configure_agents(request, get_configuration):
+    metadata = get_configuration.get('metadata')
+    agents_number = metadata['agents_number']
+    agents_os = metadata['agents_os']
+    agents_version = metadata['agents_version']
+    agents_created = create_agents(agents_number, SERVER_ADDRESS, CRYPTO, os=agents_os, version=agents_version)
+    setattr(request.module, 'agents', agents_created)
+
+
+def send_message(data_object, socket_path):
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+    sock.connect(socket_path)
+    sock.send(data_object.encode())
+    sock.close()
+
+
+def clean_logs():
+    truncate_file(LOG_FILE_PATH)
+
+
+def wait_ar_line(line):
+    if ('Active response request received: ' in line):
+        return line.split('Active response request received: ', 1)[1]
+    return None
+
+
+def validate_old_ar_message(id, message, extra_args, timeout, all_agents):
+    args = message.split()
+
+    assert args[0] == f'(local_source)', 'Event location did not match expected!'
+    assert args[1] == '[]', 'Source IP did not match expected!'
+    if all_agents == 'yes':
+        assert args[2] == 'NNS', 'AR flags did not match expected!'
+        assert args[3] in id, 'Agent ID did not match expected!'
+    else:
+        assert args[2] == 'NRN', 'AR flags did not match expected!'
+        assert args[3] == id, 'Agent ID did not match expected!'
```

Comment:
Personally I like the variable names to be representative of what they contain or represent, so that the code is easily readable. For example in this case `id` is the `message_id` (I know it can be taken out of context, but you have to do the search and that takes effort).

These are not mandatory changes but suggestions according to the style guides of this repository.